tests/testthat/test-split_data_by_task.R

```diff
@@ -0,0 +1,48 @@
+library(dplyr)
+
+# data for testing
+target_data <- readRDS(
+  testthat::test_path("testdata/flu_example_target_data.rds")
+)
+
+forecast_quantiles <- readRDS(
+  testthat::test_path("testdata/flu_example_quantile_model_output.rds")
+)
+
+valid_tbl <- validate_input_data(forecast_quantiles, target_data)
+result <- split_data_by_task(valid_tbl)
+
+split_cols <- c("horizon", "location", "target_end_date")
+
+test_that("split_data_by_task() groups data correctly", {
+  # The result is a list
+  expect_type(result, "list")
+
+  # The list length matches the number of unique combinations of split_cols
+  expect_equal(
+    length(result),
+    valid_tbl %>%
```

Comment:
consider using base R pipe `\>` in these tests, for consistency throughout the package.